addons/l10n_nl/migrations/3.3/post-migrate_update_taxes.py

```diff
@@ -22,53 +22,85 @@ def migrate(cr, version):
+        (goods_plus_tax_tag_id, goods_taxes.ids, old_plus_tax_tag_ids, goods_taxes.invoice_repartition_line_ids.ids),
+        (services_plus_tax_tag_id, services_taxes.ids, old_plus_tax_tag_ids, services_taxes.invoice_repartition_line_ids.ids),
+        (goods_minus_tax_tag_id, goods_taxes.ids, old_minus_tax_tag_ids, goods_taxes.refund_repartition_line_ids.ids),
+        (services_minus_tax_tag_id, services_taxes.ids, old_minus_tax_tag_ids, services_taxes.refund_repartition_line_ids.ids),
     ]
     insert_query_parts = []
 
-    for new_tax_tag_id, tax_ids, old_tax_tag_id, repartition_line_ids in insert_query_params:
+    for new_tax_tag_id, tax_ids, old_tax_tag_ids, repartition_line_ids in insert_query_params:
         insert_query_parts.append(
             sql.SQL("""
                 SELECT tag_aml_rel.account_move_line_id, {new_tax_tag_id}
                 FROM account_account_tag_account_move_line_rel tag_aml_rel
                 JOIN account_move_line_account_tax_rel aml_at_rel ON aml_at_rel.account_move_line_id = tag_aml_rel.account_move_line_id
                 WHERE aml_at_rel.account_tax_id = ANY({tax_ids})
-                AND tag_aml_rel.account_account_tag_id = {old_tax_tag_id}
+                AND tag_aml_rel.account_account_tag_id IN {old_tax_tag_ids}
             """).format(
                 new_tax_tag_id=sql.Literal(new_tax_tag_id),
                 tax_ids=sql.Literal(tax_ids),
-                old_tax_tag_id=sql.Literal(old_tax_tag_id),
+                old_tax_tag_ids=sql.Literal(old_tax_tag_ids),
             )
         )
 
+        if len(old_tax_tag_ids) > 1:
+            cr.execute(
+                 sql.SQL("""
+                    DELETE FROM account_account_tag_account_tax_repartition_line_rel tag_aml_rel
+                    WHERE tag_aml_rel.account_account_tag_id = {first_old_tax_tag_id}
+                    AND (
+                        SELECT COUNT(*)
+                        FROM account_account_tag_account_tax_repartition_line_rel sub_tag_aml_rel
+                        WHERE sub_tag_aml_rel.account_tax_repartition_line_id = tag_aml_rel.account_tax_repartition_line_id
+                        AND sub_tag_aml_rel.account_account_tag_id = {second_old_tax_tag_id}
+                    ) >= 1
+                """).format(
+                    first_old_tax_tag_id=sql.Literal(old_tax_tag_ids[0]),
+                    second_old_tax_tag_id=sql.Literal(old_tax_tag_ids[1]),
+                )
+            )
+
         cr.execute(
             sql.SQL("""
                 UPDATE account_account_tag_account_tax_repartition_line_rel
                 SET account_account_tag_id = {new_tax_tag_id}
                 WHERE account_tax_repartition_line_id = ANY({repartition_line_ids})
-                AND account_account_tag_id = {old_tax_tag_id}
+                AND account_account_tag_id IN {old_tax_tag_ids}
```

Comment:
@oco-odoo @lost-odoo 
This is causing syntax errors:
```suggestion
                AND account_account_tag_id = ANY({old_tax_tag_ids})
```

```
psycopg2.errors.SyntaxError: syntax error at or near "ARRAY"
LINE 5:                 AND account_account_tag_id IN ARRAY[52]
```

Reply:
I have the same problem. Can this be fixed asap? Or is there a workaround? Our customer want to go live this week.

Reply:
I am looking into it